### ISSUE TITLE
VL-BERT: Pre-training of Generic Visual-Linguistic Representations

### ISSUE THREAD

**Issue by wwacky** [wwacky/survey#27]
# 一言でいうと

# 論文情報
リンク
[OpenReview](https://openreview.net/forum?id=SygXPaEYvH)
[arxiv](https://arxiv.org/abs/1908.08530)
[code](https://github.com/jackroos/VL-BERT)

著者/所属機関
- Weijie Su (University of Science and Technology of China, Microsoft Research Asia)
- Xizhou Zhu (University of Science and Technology of China, Microsoft Research Asia)
- Yue Cao (Microsoft Research Asia)
- Bin Li (University of Science and Technology of China)
- Lewei Lu (Microsoft Research Asia)
- Furu Wei (Microsoft Research Asia)
- Jifeng Dai (Microsoft Research Asia)

投稿日付：2019-12-26 (ICLR 2020)

# 概要

# 新規性・差分

# 手法

VL-BERTは「画像のregions-of-interests(RoIs)」と「テキストのsub-word」を入力とする。RoIsは物体検出によって検出された領域か、特定のタスクでアノテーションされた領域を用いる。

入力シーケンスは[CLS]→テキスト要素→画像要素→[END]の順にする。
[SEP]はテキスト要素と画像要素の間か、異なるテキスト要素の間に挿入される。
VL-BERTはタスクに応じてシーケンスの構成が異なる。
- テキストのみの時：[CLS]→文1→[SEP]→文2→[END]
- テキストと画像の時：[CLS]→文1→[SEP]→RoIs→[END]
- テキスト2つと画像の時：[CLS]→文1→[SEP]→文2→[SEP]→RoIs→[END]

入力は以下の4つの手法でembeddingが行われる(内容は後述)。
- token embedding (BERTと同じ)
- segment embedding (BERTと同じ)
- sequence position embedding (BERTと同じ)
- visual feature embedding (VL-BERTで新たに提案)

VL-BERTの構成はFig 1の通り。

![image](https://user-images.githubusercontent.com/4422849/71499293-2ee8a400-28a3-11ea-804d-dbe86d8be69c.png)
※ Fig1. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

### token embedding
BERTにならって30,000ボキャブラリを用いてWordPiece Embeddingを行う。special tokenにはspecial embeddingが割り当てる。画像要素にはそれぞれに[IMG]のspecial tokenを割り当てる(RoIが複数ある場合はそれぞれ1つずつをtokenとして扱い、[IMG]を割り当てるみたい)。

### Visual Feature Embedding
visual appearance featureとvisual geometry embeddingを結合した全結合層の出力を用いる。
visual appearance featureには、各RoIに対応するFast R-CNN detectorの出力層の手前の特徴ベクトル(本論文では2048次元)を利用する。画像要素以外に対しては、画像全体をRoIとしてFast R-CNNを用いて画像全体からvisual appearance featureを抽出する。画像要素以外には[CLS]などのspecial tokenも含まれる。

visual geometry embeddingは、入力画像の位置情報をVL-BERTに伝えるために設計されたもの。各RoIはRoIのtop-leftとbottom-rightの座標を画像の幅と高さで正規化された4次元ベクトルで特徴付けられる。Relation Networkでの実装を参考に、この4次元ベクトルをsin, cos関数を用いて2048次元に埋め込む。

### Segment Embedding
A,B,Cの3種類のsegmentを利用する。Aは1つ目のテキスト、Bは2つ目のテキスト、Cは画像要素に割り当てられる。
例えば、<Question, Answer, Image>が入力の場合は、QuestionにはA, AnswerにはB、ImageにはCが割り当てられる。<Caption, Image>が入力の場合はCaptionにA、ImageにCが割り当てられる。

### Sequence Embedding
BERT同様に、各要素の順序を学習させる。画像要素には自然な順序は存在しないため、どの順序で入力しても同じ結果が得られるようにする。そのため、画像要素のposition embeddingは全て同じ値になる(Fig 1を参照すると分かりやすい)

## pre-training
visual-linguisticデータセットとtext-onlyデータセットでpre-trainingを行う。
visual-linguisticデータセットにはConceptual Captionsを用いる。これには330万のキャプション付き画像が含まれる。
text-only datasetにはBERTのpre-trainingにも利用されているBooksCorpusとEnglish Wikipediaデータセットを利用する。
SGDのミニバッチでは、Conceptual CaptionsとBooksCorpus&English Wikipediaから1:1の比率でランダムサンプリングしたデータを用いる。Conceptual Captionsに対しては、Faster R-CNNを用いてRoIsの特定とカテゴリ分類を行い、後述のpre-training用のタスクに用いる。

### Task 1: Masked Language Modeling with Visual Clues
BERTのMasked Language Modeling (MLM)と似たタスク。違いは画像特徴量が手がかりとして利用できるところ。入力文の各単語が約15%の確率で[MASK]に置き換えられマスクされる。マスクされた単語を、マスクされていない単語と画像特徴量から予測する。これによってテキスト間の依存関係だけではなく、テキストと画像の対応も得られる。
例えば、Fig 1で"kitten drinking from [MASK]"で、入力画像なしではマスクされた単語は"bowl"、"spoon"、"bottle"など複数のものが当てはまる。画像を手がかりにすれば"bottle"と画像のRoIsを対応させて正しい推測を行う。全ボキャブラリに対するクラス分類で、Softmax cross-entropy lossで学習を行う。

BooksCorpusとEnglish Wikipediaから得られたサンプルでpre-trainingを行う場合は、入力フォーマットを画像を含まない<Text, 空集合>として扱う。lossはBERTのMLMと同じ。

### Task 2: Masked RoI Classification with Linguistic Clues
各RoIを約15%の確率でランダムにマスクし、その他の情報からマスクされたRoIのカテゴリを予測する。RoIのカテゴリはFast R-CNNを適用して得られたカテゴリを正解とする。leakageを避けるために、マスクされたRoIに対応する箇所のピクセルを0にセットしてからFast R-CNNを適用する。
図1の例だと、マスクされたRoIのカテゴリを画像情報から予測することはできないが、キャプションの"kitten drinking from bottle"を手がかりとすれば予測することができる。
  
  
  
以下はpre-trainingされたVL-BERTのattentionを可視化したもの。

![image](https://user-images.githubusercontent.com/4422849/71499467-001efd80-28a4-11ea-95af-f935f3aa4219.png)
![image](https://user-images.githubusercontent.com/4422849/71499474-06ad7500-28a4-11ea-8d07-494c594b716b.png)
![image](https://user-images.githubusercontent.com/4422849/71499483-0dd48300-28a4-11ea-9384-b5bade397865.png)
※ Fig3. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

# 結果

## fine-tuning on downstream tasks
入力フォーマット、output prediction、loss関数、学習方法を変更することで、様々なタスクに対するfine-tuningを行うことができる。

### VISUAL COMMONSENSE REASONING (VCR)
画像とカテゴリ付きのRoIのリストが与えられた状態で質問が与えられる。モデルは質問に対する回答と根拠を予測する。各質問に対して、4つの回答の選択肢と、4つの根拠の選択肢から選択する。
タスクQ→ARは2つのサブタスクに分解される。question answering(Q→A)と、answer justification(QA→R)の2つ。
VCRデータセットは26.5万の質問、回答、根拠の組み合わせと10万画像(動画のシーン)が含まれている。training, validation, testはそれぞれ21.3万質問と8万画像、2.7万質問と1万画像、2.5万質問と1万画像に分割する。
VL-BERTでは入力に<Question, Answer, Image>を用いる。[CLS]の出力から与えられた回答のどれが正しいかSoftmaxでクラス分類を行う。RoIの入力にはデータセットに含まれている正解を用いる。fine-tuneでは2つのlossを用い、正しい回答のクラス分類ができたかどうかと、RoIのクラス分類を行う。

![image](https://user-images.githubusercontent.com/4422849/71536789-97448d80-2956-11ea-9584-5e54d61dcc41.png)
※ Fig2. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

Table 1からVL-BERTがSOTAを達成している事がわかる。

![image](https://user-images.githubusercontent.com/4422849/71499317-52135380-28a3-11ea-8b01-9d380f459100.png)
※ Table1. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

#### VISUAL QUESTION ANSWERING (VQA)

VQAでは生の画像と、画像に基づく質問が与えられる。モデルはこれに対して正しい回答を生成or選択を行う。ここではVQA v2.0データセットを利用し、モデルはそれぞれの質問に対応する回答を共通する3,129個の共有セットの中から選択する。
VL-BERTで行う際は<Question, Answer, Image>のスキーマを用いる。ただし、Answerの箇所は[MASK]トークンを与える。RoIはFast R-CNNで抽出された領域を与える。学習は、取りうる回答に対するmulti-class cross-entropy lossを用いる。

![image](https://user-images.githubusercontent.com/4422849/71537028-9c570c00-2959-11ea-9cea-4b4a7005b4ee.png)
※ Fig2. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

VQAの性能はVL-BERTとLXMERTの性能が高いが、LXMERTは大量のvisual question answeringデータでpre-trainingを行っているためである。対して、VL-BERTはConceptual Captionsのみでpre-trainingを行っている。
(Table 2のVL-BERT without pre-trainingとVL-BERT_BASEのtest-stdが記録されていない理由は不明)

![image](https://user-images.githubusercontent.com/4422849/71499334-5fc8d900-28a3-11ea-9dbe-dae9bd054526.png)
※ Table2. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

### REFERRING EXPRESSION COMPREHENSION
referring expressionとは、画像内の物体について自然言語で言及することである。referring expression comprehension タスクは、自然言語の画像に対する表現が与えられ、それに対応する物体領域を特定するものである。
評価はRefCOCO+データセットで行う。RefCOCO+では、testAに複数の人物が写っている画像が含まれており、testBの方には複数の物体が写っている画像が含まれている。
VL-BERTでは入力フォーマットに<Query, Image>を用いる。モデルの学習には、正解のRoIsか、MAttNetで検出された領域を用いる。学習では、入力のRoIsに対するクラス分類のスコアを計算する。各RoIに対して、binary classification lossを適用する。推論では、最も高いスコアのRoIを選択する。

![image](https://user-images.githubusercontent.com/4422849/71537177-a24dec80-295b-11ea-8eaa-de8c917cad3a.png)
※ Fig2. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

Table 3のGround-truth Regionsはデータセット内に含まれるRoIを用いた場合の評価で、Detected RegionsはMAttNetで検出された領域のどれが正解かを評価したものと思われる（恐らく、MAttNet・ViLBERTと評価を揃えるため）

![image](https://user-images.githubusercontent.com/4422849/71499356-71aa7c00-28a3-11ea-9a97-8b02e4e5df60.png)
※ Table3. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

## pre-trainingの効果検証

pre-trainingの効果を検証するために、pre-trainingのタスクを調整して精度比較を行った。Table 4参照。fine-tuningのエポック数は半分にしている(特に理由なし。実験時間の短縮？)。

![image](https://user-images.githubusercontent.com/4422849/71499367-80912e80-28a3-11ea-986f-803d7954bbab.png)
※ Table4. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

内容は以下の通り
- w/o pre-trainingの精度が低いので、pre-trainingの効果がある
- (a)(b)からpre-trainingのTask 1とTask 2はRefCOCO+には効果があるが、VCRとVQRにはあまり効果がない
    - これはRefCOCO+のみが予測を[IMG]トークンに対応するように行っているためであると思われる
- (c)ではViLBERTとLXMERTで行われているSentence-Image Relationship Prediction(画像とキャプションの対応が正しいかの予測)を行っているが、精度が下がっている
    - これは画像とキャプションが不一致の場合がネガティブサンプルになるためだと思われる
- (d) ではテキストのみの学習を行っているが、VCRで精度が向上している
    - VCRは複雑で長い文章が含まれているためだと思われる
- Fast R-CNNのパラメータをpre-trainingで固定せずに更新すると、更に精度が上がった。これをVL-BERT_BASEとした

### VideoBERT
lisual-linguisticのpretrainingを最初に扱った研究。ビデオクリップをaction recognition用の既存ネットワークを用いて加工し、得られた特徴に基つく異なるクラスタ(visual words)に割り当てる。
しかし、ビデオクリップのクラスタリングは画像の内容のかなりを減らし、visual networkのパラメータ更新を妨げてしまう。これに続くCBTではクラスタリングの処理は取り除かれている。VideoBERT、CBTともに動画に対するもので、時間方向に対して線形構造をとっている。

### VilBERT, LXMERT
VilBERT, LXMERTのどちらとも、文章と画像のそれぞれ別の入力とする2つのsingle-modal networkを採用し、それに続くcross-modal Transformerで2つのモーダルの情報を組み合わせている。cross-modal Transformer内でのattentionパターンは制限されている。
VilBERTの著者はsingle-streamに統一されたモデルよりも、two-streamモデルの方が優れていると主張している。提案手法のVL-BERTは、attentionパターンに制限がないTransformerに基づく統一構造を採用している。そのため、文章と画像は早く・自由に相互作用を行うようになる。

### VisualBERT, B2T2, Unicoder-VL
これらはまだレビュー中だが、統一されたsingle-stream構造を採用している。これらの手法の違いはTable 5に示す。

![image](https://user-images.githubusercontent.com/4422849/71499390-9c94d000-28a3-11ea-82cb-8026fd4bab8e.png)
※ Table5. VL-BERT: Pre-training of Generic Visual-Linguistic Representationsより引用

加えて、VL-BERTとその他のpretrainingの手法の注目すべき3つの違いを挙げる(これらの効果はSection4.3で示す)
1. 他の全てのタスクで利用されているSentence-Image Relationship Predictionは、visual-linguistic representationのpretrainingの役に立たない。そのため、VL-BERTではこのタスクは採用していない
1. VL-BERTではvisual-linguistic datasetとtext-only datasetの両方でpretrainingを行っている。このようなjoint pretrainingは長く複雑な文章の一般化を向上させることが分かっている。
1. visual representationのチューニングを改善している。VL-BERTではpre-training時に画像特徴量を作るFast R-CNNの重みを更新している。Masked RoI Classificationでのvisual clue leakageを避けるために、convolution層ではなく入力の生のピクセルに対してマスク処理を行っている

## 略語
頭字語 | 意味
--- | ---
RoI | region-of-interest
